includes/device-groups.inc.php

```diff
@@ -214,13 +214,22 @@ function GetDevicesFromGroup($group_id, $nested = false, $full = '')
 }//end GetDevicesFromGroup()
 
 /**
- * Get all Device-Groups
+ * Get all the Device groups which the decive are associated to the current user
  * @return array
  */
 function GetDeviceGroups()
 {
-    return dbFetchRows('SELECT * FROM device_groups ORDER BY name');
-}//end GetDeviceGroups()
+    $user_id = '';
```

Comment:
Have you tried this instead of all the code in this function?:
```
return \App\Models\DeviceGroup::hasAccess(\Auth::user())->get()->toArray();
```

Reply:
You code is smarter than mine, but same wrong effect.
For some reason, a normal user can see as many duplicate menu items under Devices / Device groups, as many device belongs to each group.
I tried also to add a GROUP BY to my query, but the effect is the same.
Which PHP file manages the menu composition?

Reply:
app\Models\DeviceGroup.php   Looks like the code there is still using a join... which is why you get duplicates.  I'll try to fix that.

Reply:
Also to fetch data for the menu is `MenuComposer.php` and the template is `menu.blade.php`